apriltag_ros/src/common_functions.cpp

```diff
@@ -793,4 +829,61 @@ bool TagDetector::findStandaloneTagDescription (
   return true;
 }
 
+bool TagDetector::isTagInFOV(const tf::Transform& tag_pose) const
+{
+  if (!camera_model_.initialized())
+  {
+    ROS_WARN_STREAM_THROTTLE(5.0, "Camera not initialized, skipping FOV check");
+    return false;
+  }
+  // convert the 3d pose in the image frame to a 2d pixel frame
+  if (tag_pose.getOrigin().getZ() > 0)
```

Comment:
```suggestion
  if (tag_pose.getOrigin().getZ() <= 0)
  {
    return false;
  }
```

instead of putting the other compute inside the if